src/components/pages/users.js

```diff
@@ -26,38 +31,201 @@ const availableRoles = Object.values(ROLE);
+    const newStorage = {
+      ...storageBlueprint,
+      plan: newStoragePlanId
+    };
+
+    setIsSaving(true);
+    setUsers(oldUsers => replaceItem(oldUsers, { ...user, storage: newStorage }));
+
+    let finalStorage;
+    try {
+      finalStorage = await userApiClient.saveUserStoragePlan({ userId: user._id, storagePlanId: newStoragePlanId });
+    } catch (error) {
+      errorHelper.handleApiError({ error, logger, t });
+      finalStorage = oldStorage;
+    } finally {
+      setUsers(oldUsers => replaceItem(oldUsers, { ...user, storage: finalStorage }));
+      setIsSaving(false);
+    }
+  };
+
+  const handleAddReminderClick = async user => {
+    const oldStorage = user.storage;
+
+    const storageBlueprint = oldStorage || getDefaultStorage();
+    const newStorage = {
+      ...storageBlueprint,
+      reminders: [
+        ...storageBlueprint.reminders,
+        {
+          timestamp: new Date().toISOString(),
+          createdBy: executingUser._id
+        }
+      ]
+    };
+
+    setIsSaving(true);
+    setUsers(oldUsers => replaceItem(oldUsers, { ...user, storage: newStorage }));
+
+    let finalStorage;
+    try {
+      finalStorage = await userApiClient.addUserStorageReminder({ userId: user._id });
+    } catch (error) {
+      errorHelper.handleApiError({ error, logger, t });
+      finalStorage = oldStorage;
+    } finally {
+      setUsers(oldUsers => replaceItem(oldUsers, { ...user, storage: finalStorage }));
+      setIsSaving(false);
+    }
+  };
+
+  const handleRemoveRemindersClick = async user => {
+    const oldStorage = user.storage;
+
+    const storageBlueprint = oldStorage || getDefaultStorage();
+    const newStorage = {
+      ...storageBlueprint,
+      reminders: []
+    };
+
+    setIsSaving(true);
+    setUsers(oldUsers => replaceItem(oldUsers, { ...user, storage: newStorage }));
+
+    let finalStorage;
+    try {
+      finalStorage = await userApiClient.removeUserStorageReminders({ userId: user._id });
+    } catch (error) {
+      errorHelper.handleApiError({ error, logger, t });
+      finalStorage = oldStorage;
+    } finally {
+      setUsers(oldUsers => replaceItem(oldUsers, { ...user, storage: finalStorage }));
+      setIsSaving(false);
+    }
+  };
+
+  const handleDeleteAllPrivateRoomsClick = user => {
+    confirmAllPrivateRoomsDelete(t, user.username, async () => {
+      const oldStorage = user.storage;
+
+      const storageBlueprint = oldStorage || getDefaultStorage();
+      const newStorage = storageBlueprint.plan
+        ? {
+          ...storageBlueprint,
+          usedStorageInBytes: 0,
+          reminders: []
+        }
+        : null;
+
+      setIsSaving(true);
+      setUsers(oldUsers => replaceItem(oldUsers, { ...user, storage: newStorage }));
+
+      let finalStorage;
+      try {
+        await roomApiClient.deleteAllPrivateRoomsForUser({ ownerId: user._id });
+        finalStorage = await userApiClient.removeUserStorageReminders({ userId: user._id });
```

Comment:
In the next step regarding storage, I'm thinking we might want to do the room deletion in a transaction 🤔 Maybe we can introduce that when we introduce the user lock.

Reply:
Yes, I also think this should happen within one call/transaction, but at the moment we manage the rooms with the `roomClient/Service` and the storage reminders with `userClient/Service`. I think we already agreed to factor out a `storageClient/Service` in order to disentangle these dependencies.

Reply:
Yes exactly, I only mentioned thi to we keep it in mind as well.